File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        subarray, avg = sum(arr[:k]), k * threshold
        ans = 1 if subarray >= avg else 0
        
        for i in range(len(arr)-k):
            subarray += arr[k+i] - arr[i]
            ans = ans + 1 if subarray >= avg else ans
            
        return ans

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& nums, int k, int threshold) {
        int n = nums.size(), cur = 0, ans = 0;
        for (int i = 0; i < n; i++) {
            cur += nums[i];
            if (i == k - 1)
                ans += (cur / k >= threshold);
            if (i >= k) {
                cur -= nums[i - k];
                ans += (cur / k >= threshold);
            }
        }
        return ans;
    }
};